engines/python/setup/djl_python/properties_manager/vllm_rb_properties.py

```diff
@@ -45,12 +45,19 @@ class VllmRbProperties(Properties):
     draft_model_tp_size: int = 1
     record_acceptance_rate: Optional[bool] = False
 
-    @validator('engine')
-    def validate_engine(cls, engine):
-        if engine != "Python":
+    @root_validator(skip_on_failure=True)
+    def validate_engine(cls, properties):
+        engine = properties["engine"]
+        rolling_batch = properties["rolling_batch"]
+        if rolling_batch == RollingBatchEnum.vllm and engine != "Python":
             raise AssertionError(
                 f"Need python engine to start vLLM RollingBatcher")
-        return engine
+
+        if rolling_batch == RollingBatchEnum.lmidist_v2 and engine != "MPI":
```

Comment:
Could we just have individual parameter checking for each rolling batch implementation?